File: src/main/java/se/clouds/app/javanet/app/Controllers/WeatherForecastController.java
package se.clouds.app.javanet.app.Controllers;

import se.clouds.app.javanet.app.domain.weatherforecast.query.GetWeatherForecast;
import se.clouds.app.javanet.core.controller.AppController;
import se.clouds.app.javanet.core.controller.IActionResult;
import se.clouds.app.javanet.core.di.Di;
import se.clouds.app.javanet.core.mediator.MediatR;

public class WeatherForecastController extends AppController
{
    private MediatR<IActionResult> mediatr = (MediatR)Di.GetMediator();
    private static String routePath = "/weatherforecast";

    public WeatherForecastController()
    {
        super(routePath);
    }

    public IActionResult Get()
    {
        var result = mediatr.SendRequest(new GetWeatherForecast());
        return result.orElseThrow();
    }
}

File: Components/app/src/main/dotnet/app/controllers/WeatherForecastController.cs
using System.Threading;
using System.Threading.Tasks;
using MediatR;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using se.clouds.javanet.app.domain.weatherforecast;
using se.clouds.javanet.app.domain.weatherforecast.queries;

namespace se.clouds.javanet.app.controllers
{
    [ApiController]
    [Route("[controller]")]
    public class WeatherForecastController : ControllerBase
    {
        private readonly IMediator _mediator;
        private readonly ILogger<WeatherForecastController> _logger;
        public WeatherForecastController(IMediator mediator, ILogger<WeatherForecastController> logger)
        {
            _mediator = mediator;
            _logger = logger;
        }

        [HttpGet]
        public async Task<WeatherForecast[]> GetAsync()
        {
            return await _mediator.Send(new GetWeatherForecast(), new CancellationToken());
        }
    }
}
